src/prefect/runner/_observers.py

```diff
@@ -36,12 +39,16 @@ def __init__(
 
         Args:
             on_cancelling: Callback to call when a flow run is marked as cancelling.
-            flow_run_ids: List of flow run IDs to poll when websocket connection is lost.
             polling_interval: Interval in seconds to poll for cancelling flow runs when websocket connection is lost.
+            event_filter: Optional event filter to use for the websocket subscription.
+                If not provided, defaults to filtering for "prefect.flow-run.Cancelling" events.
         """
         self.logger = get_logger("FlowRunCancellingObserver")
         self.on_cancelling = on_cancelling
         self.polling_interval = polling_interval
+        self._event_filter = event_filter or EventFilter(
+            event=EventNameFilter(name=["prefect.flow-run.Cancelling"])
+        )
```

Comment:
We probably want some validation to ensure the passed event filter at the very least includes the `prefect.flow-run.Cancelling` event name to avoid a potential footgun. 

Other options include updating the provided filter to include that event name or allowing only certain portions of an event filter to be passed in.

Reply:
Good call! You could pass a less restrictive filter and end up cancelling more than intended